features/topology-map/plugins/org.opennms.features.topology.plugins.topo.graphml/src/main/java/org/opennms/features/topology/plugins/topo/graphml/internal/GraphMLMetaTopologyFactory.java

```diff
@@ -124,10 +126,39 @@ public void updated(String pid, @SuppressWarnings("rawtypes") Dictionary propert
 
 					// Vertex Status Provider
 					// Only add status provider if explicitly set in GraphML document
-					if (rawTopologyProvider.requiresStatusProvider()) {
-						GraphMLVertexStatusProvider statusProvider = new GraphMLVertexStatusProvider(
-								rawTopologyProvider.getVertexNamespace(),
-								(nodeIds) -> m_serviceAccessor.getAlarmDao().getNodeAlarmSummariesIncludeAcknowledgedOnes(nodeIds));
+					final StatusProvider statusProvider;
+					switch (rawTopologyProvider.vertexStatusProvider()) {
```

Comment:
Can we put this into a method `getStatusProvider(rawTopologyProvider.getVertextStatusProviderType())`?